### ISSUE TITLE
Plugin not quite getting right data for Specials

### ISSUE THREAD

**Issue by ReinT0** [jwueller/jellyfin-plugin-onepace#12]
It doesn't seem like it is fetching metadata for the specials folder quite correctly, some get cover arts, and some of them don't. The Gaimon April Fools special and the Strong World special don't even get episode names:
[log.txt](https://github.com/jwueller/jellyfin-plugin-onepace/files/11938756/log.txt)
https://prnt.sc/k5cslMN8p806
https://prnt.sc/RwXSxeooAgfR

The Whole Cake Island April fools special get placed in the Whole Cake Island Arc folder, while also being in the specials folder.
https://prnt.sc/WGy_pSPgDPPG
https://prnt.sc/RD1nDNOoX6hA

So should add a little verifyer that insures that the metadata for that folder is getting the right metadata and getting placed where it belongs. I believe your scripts searches for the episode name, and then places it in which arc it belongs. Some lines that verify that it is in the correct folder and then places it in right arc might be a good edit to the script. But as far I know this error is only for April fools special since the episode name is the same as the original.

**Comment by jwueller** [comment]
Thanks for typing this up again! I'm coming back from my vacation, so I will try to take a look at this sometime tomorrow.

**Comment by jwueller** [comment]
I managed to reproduce this locally.

**Comment by jwueller** [comment]
It turns out that the main cause of this is that the filenames don't match up cleanly with the API specials episode info.

Here are some filename examples:
```
[One Pace][0] Strong World [480p][76B2C869].mkv
[One Pace][823-824] Whole Cake Island 01 [720p][EDB72EE5].mp4
[One Pace][236-262 Covers] Wapol's Omnivorous Hurrah [720p][05BE81E6].mkv
[One Pace] Straw Hat Theatre [720p][415455AE].mkv
```

But these are known as `Special 01`, `Special 02`, etc. in the API.

This will probably has to be postponed until the API provides the CRC32 to compare against, since that's the only unambiguous identifier for some of these filenames.

**Comment by jwueller** [comment]
CRC-32 filename matching is now implemented in v1.3.2.0! I'm going to wait for your feedback before I close this, if you get a chance to try it out @ReinT0.

**Comment by jwueller** [comment]
What is the filename for that one?

**Comment by jwueller** [comment]
@ReinT0 Did you get the Gaimon special working?

**Comment by ReinT0** [comment]
Sorry I have been busy with moving and stuff for Uni, so my server have been offline, will give you reply when I have it online and looked into the issue again.

**Comment by jwueller** [comment]
Interesting, this seems to be related to the exception from the log in #28. Unfortunately that error message from `System.Text.Json` is pretty bad, so I will have to try to reproduce this locally.

**Comment by jwueller** [comment]
This should be solved in 1.4.0.0.
